sdv/data_processing/data_processor.py

```diff
@@ -332,7 +332,7 @@ def _validate_constraint_dict(self, constraint_dict):
                     'to columns that are part of an address group.'
                 )
 
-        constraint_class._validate_metadata(self.metadata, **constraint_parameters)
+        constraint_class._validate_metadata(**constraint_parameters)
```

Comment:
shouldn't all the validate methods be crashing now because they examine the metadata object that was previously passed in? eg.
https://github.com/sdv-dev/SDV/blob/f25037ba6b7f48ffb86a53745ac57bc900e7af6e/sdv/constraints/base.py#L140-L146